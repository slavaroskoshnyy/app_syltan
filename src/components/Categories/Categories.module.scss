.wrapper {
   display: flex;
   margin: 0 -5.5px;
   padding-bottom: 30px;
}