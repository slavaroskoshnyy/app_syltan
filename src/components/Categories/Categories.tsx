import React from "react";
import CardCategories from "../UI/cardCategories/CardCategories";
import cl from "./Categories.module.scss";

export default function Categories() {
  const categories = [
    { name: "Уход за телом" },
    { name: "Уход за руками" },
    { name: "Уход за ногами" },
    { name: "Уход за лицом" },
    { name: "Уход за волосами" },
    { name: "Средства для загара" },
    { name: "Средства для бритья" },
    { name: "Подарочные наборы" },
    { name: "Гигиеническая продукция" },
    { name: "Гигиена полости рта" },
    { name: "Бумажная продукция" },
  ];
  return (
    <div className={cl.wrapper} role="text">
      {categories.map((elem) => (
        <CardCategories name={elem.name} key={elem.name} />
      ))}
    </div>
  );
}
