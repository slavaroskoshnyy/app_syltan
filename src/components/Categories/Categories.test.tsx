import { render, screen } from "@testing-library/react";

import Categories from "./Categories";

describe("Categories component", () => {
  it("Categories renders", () => {
    render(<Categories />);

    expect(screen.getByRole("text")).toBeInTheDocument();
    expect(screen.getByText("Уход за телом")).toBeInTheDocument();
    expect(screen.getByText("Средства для загара")).toBeInTheDocument();
    expect(screen.getByText("Бумажная продукция")).toBeInTheDocument();
  });
  it("Categories snapshot", () => {
    const view = render(<Categories />);

    expect(view).toMatchSnapshot();
  });
});
