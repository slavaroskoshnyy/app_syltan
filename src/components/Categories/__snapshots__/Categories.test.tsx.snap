// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component Categories snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
        role="text"
      >
        <div
          class="text"
        >
          <div>
            <p>
              Уход за телом
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Уход за руками
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Уход за ногами
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Уход за лицом
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Уход за волосами
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Средства для загара
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Средства для бритья
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Подарочные наборы
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Гигиеническая продукция
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Гигиена полости рта
            </p>
          </div>
        </div>
        <div
          class="text"
        >
          <div>
            <p>
              Бумажная продукция
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
      role="text"
    >
      <div
        class="text"
      >
        <div>
          <p>
            Уход за телом
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Уход за руками
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Уход за ногами
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Уход за лицом
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Уход за волосами
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Средства для загара
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Средства для бритья
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Подарочные наборы
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Гигиеническая продукция
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Гигиена полости рта
          </p>
        </div>
      </div>
      <div
        class="text"
      >
        <div>
          <p>
            Бумажная продукция
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
